remote-api.js

```diff
@@ -29,18 +25,41 @@ function recurse (obj, manifest, path, remoteCall) {
 }
 
 
-module.exports = function (obj, manifest, _remoteCall, bootstrap) {
+function noop (err) {
+  if (err) {
+    throw explain(err, 'callback not provided')
+  }
+}
 
+module.exports = function (obj, manifest, _remoteCall, bootstrap) {
   obj = obj || {}
 
   function remoteCall(type, name, args) {
-    var cb = isFunction (args[args.length - 1]) ? args.pop() : noop
+    var cb = isFunction (args[args.length - 1])
+      ? args.pop()
+      : type.endsWith('sync')
```

Comment:
can the condition be `type === 'async' || type === 'sync'`? I think it's safer, for various reasons. (1) when reading code quickly, you can think that this is only about 'sync', because the string 'sync' is there, (2) same thing can happen if you search the codebase for 'sync', or if you search the codebase for 'async' and this line never shows up, (3) what if we add `'___sync'` types in the future? This condition would automatically catch that new type.

Reply:
Good call! I'll add a commit to take care of that now.